### ISSUE TITLE
VkFramebufferCreateInfo correct aspectMask unspecified?

### ISSUE THREAD

**Issue by neXyon** [KhronosGroup/Vulkan-Docs#1469]
With my Nvidia card I can use a stencil buffer in a render pass of which the framebuffer was created with a `VkImageView` that lacks the `VK_IMAGE_ASPECT_STENCIL_BIT` in its `aspectMask`. I found no limitation in the specification that explicitly states that the bit has to be set in order to be able to use the stencil buffer.

It seems like on AMD, this wrong implementation can cause issues however: https://stackoverflow.com/a/36162230

Certainly, I currently don't get any warnings from the validation layers.

Maybe there should be explicit mention about the `aspectMask` of a `VkImageView` in the *Valid Usage* section of `VkFramebufferCreateInfo`?

**Comment by sfricke-samsung** [comment]
Just to be clear, when you say `stencil buffer` what are you refering to?

A `VkImage` with `VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT` and a format like `VK_FORMAT_S8_UINT`?

----

Looking for where/if it would be validated, at `vkCreateRenderPass` time, you use `pDepthStencilAttachment` and if you are using a layout such as `VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL` that is all valid I think

Pipeline creation time is where you set `stencilTestEnable` which would enable [Stencil testing](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#fragops-stencil)

So you wouldn't know until a `vkCmdDraw*` command when you have both the framebuffer (from the renderpass) and the pipeline bound that you are trying to do stencil testing with a framebuffer that doesn't have a `VK_IMAGE_ASPECT_STENCIL_BIT` imageView in the `pDepthStencilAttachment`

...I am not sure what the correct behavior is for this case, but at least draw time is where it would be validated if its invalid app behavior

**Comment by neXyon** [comment]
Yes, that's basically what I mean with a stencil buffer. I assume if it's really a stencil only format (like `VK_FORMAT_S8_UINT`) the creation of the `VkImageView` should already require `VK_IMAGE_ASPECT_STENCIL_BIT`, since that's specified under `VkImageSubresourceRange` but for multiplane formats such as `VK_IMAGE_ASPECT_STENCIL_BIT` and a `VkImage` with `VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT` you may choose to use just one of `VK_IMAGE_ASPECT_DEPTH_BIT` and `VK_IMAGE_ASPECT_STENCIL_BIT`.

You're right, it couldn't be validated before the connection between the pipeline and the framebuffer would be made, maybe during the `vkCmdBindPipeline` call after a corresponding `vkCmdBeginRenderPass`, if you want to detect it before `vkCmdDraw*`?

**Comment by oddhack** [comment]
This should be fixed in the 1.4.326 spec update.
